src/interface/viewport.c

```diff
@@ -2632,13 +2634,321 @@ void store_interaction_info(paint_struct *ps)
+
+	sint16 height;
+	uint8 *new_source_pointer_start, *esi_end;
+
+	if (dpi->zoom_level != 0) {
+		if (image->flags & 0x20) {
+			return false;
+		}
+
+		if (image->flags & 0x10) {
+			// TODO: SAR in dpi done with `>> 1`, in coordinates with `/ 2`
+			rct_drawpixelinfo zoomed_dpi = {
+					.bits = dpi->bits,
+					.x = dpi->x >> 1,
+					.y = dpi->y >> 1,
+					.height = dpi->height,
+					.width = dpi->width,
+					.pitch = dpi->pitch,
+					.zoom_level = dpi->zoom_level - 1
+			};
+
+			return new_sub_679074(&zoomed_dpi, imageId - image->zoomed_offset, x / 2, y / 2);
+		}
+	}
+
+	RCT2_GLOBAL(0x9E3D08, uint8*) = image->offset;
+	RCT2_GLOBAL(0x9E3D0C, sint16) = image->width;
+	RCT2_GLOBAL(0x9E3D0C + 2, sint16) = image->height;
+	RCT2_GLOBAL(0x9E3D10, sint16) = image->x_offset;
+	RCT2_GLOBAL(0x9E3D10 + 2, sint16) = image->y_offset;
+	RCT2_GLOBAL(0x9E3D14, uint16) = image->flags;
+	RCT2_GLOBAL(0x9E3D14 + 2, uint16) = image->zoomed_offset;
+
+	int round;
+	switch (dpi->zoom_level) {
+		case 0:
+		default:
+			round = 1;
+			break;
+
+		case 1:
+			round = 2;
+			break;
+
+		case 2:
+			round = 4;
+			break;
+
+		case 3:
+			round = 8;
+	}
+
+	if (image->flags & G1_FLAG_RLE_COMPRESSION) {
+		y -= (round - 1);
+	}
+
+	y += image->y_offset;
+	sint16 yStartPoint = 0;
+	height = image->height;
+	if (dpi->zoom_level != 0) {
+		if (height % 2) {
+			height--;
+			yStartPoint++;
+		}
+
+		if (dpi->zoom_level == 2) {
+			if (height % 4) {
+				height -= 2;
+				yStartPoint += 2;
+			}
+		}
+
+		if (height == 0) {
+			return false;
+		}
+	}
+
+	y = floor2(y, round);
+	sint16 yEndPoint = height;
+	y -= dpi->y;
+	if (y < 0) {
+		yEndPoint += y;
+		if (yEndPoint <= 0) {
+			return false;
+		}
+
+		yStartPoint -= y;
+		y = 0;
+	}
+
+	y += yEndPoint;
+	y--;
+	if (y > 0) {
+		yEndPoint -= y;
+		if (yEndPoint <= 0) {
+			return false;
+		}
+	}
+
+	sint16 xStartPoint = 0;
+	sint16 xEndPoint = image->width;
+	if (!(image->flags & G1_FLAG_RLE_COMPRESSION)) {
+		RCT2_GLOBAL(0x009ABDAE, sint16) = 0;
+	}
+
+	x += image->x_offset;
+	x = floor2(x, round);
+	x -= dpi->x;
+	if (x < 0) {
+		xEndPoint += x;
+		if (xEndPoint <= 0) {
+			return false;
+		}
+
+		if (!(image->flags & G1_FLAG_RLE_COMPRESSION)) {
+			RCT2_GLOBAL(0x009ABDAE, sint16) -= x;
+		}
+
+		xStartPoint -= x;
+		x = 0;
+	}
+
+	x += xEndPoint;
+	x--;
+	if (x > 0) {
+		xEndPoint -= x;
+		if (xEndPoint <= 0) {
+			return false;
+		}
+
+		if (!(image->flags & G1_FLAG_RLE_COMPRESSION)) {
+			RCT2_GLOBAL(0x009ABDAE, sint16) += x;
+		}
+	}
+
+	if (image->flags & G1_FLAG_RLE_COMPRESSION) {
+		return sub_67933B_679788_679C4A_67A117(image->offset, xStartPoint, yStartPoint, round);
+	}
+
+	uint8 *esi = image->offset + (yStartPoint * image->width) + xStartPoint;
+	uint32 ebx = RCT2_GLOBAL(0x00EDF81C, uint32);
+
+	if (!(image->flags & 2)) {
+		return sub_679236_679662_679B0D_679FF1(ebx, image, esi);
+	}
+
+	loc_6791B8_6795E4_679A8F_679F73(image, esi, &new_source_pointer_start, &esi_end);
```

Comment:
I've never tested anything that has made it to this point. Do you have an image that does?